modules/unsupported/cql-json/src/main/java/org/geotools/filter/text/cqljson/FilterToCQL2Json.java

```diff
@@ -0,0 +1,543 @@
+            List<Expression> parameters = function.getParameters();
+            Expression param3 = parameters.get(2);
+            if (param3 instanceof Literal) {
+                Literal literal = (Literal) param3;
+                Object value = literal.getValue();
+                if (!(value instanceof String)) {
+                    return false; // not a relate
+                }
+            }
+        } else {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public Object visit(PropertyIsNotEqualTo filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        return buildComparison(filter, output, "<>");
+    }
+
+    @Override
+    public Object visit(PropertyIsGreaterThan filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        return buildComparison(filter, output, ">");
+    }
+
+    @Override
+    public Object visit(PropertyIsGreaterThanOrEqualTo filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        return buildComparison(filter, output, ">=");
+    }
+
+    @Override
+    public Object visit(PropertyIsLessThan filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        return buildComparison(filter, output, "<");
+    }
+
+    @Override
+    public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        return buildComparison(filter, output, "<=");
+    }
+
+    @Override
+    public Object visit(PropertyIsLike filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        output.put(OP, "like");
+
+        ArrayNode args = objectMapper.createArrayNode();
+        Expression expr = filter.getExpression();
+        expr.accept(new ExpressionToCQL2Json(objectMapper), args);
+        final String pattern = filter.getLiteral();
+        args.add(pattern);
+        output.set(ARGS, args);
+
+        return output;
+    }
+
+    @Override
+    public Object visit(PropertyIsNull filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        output.put(OP, "isNull");
+
+        ArrayNode args = objectMapper.createArrayNode();
+        Expression expr = filter.getExpression();
+        expr.accept(new ExpressionToCQL2Json(objectMapper), args);
+        output.set(ARGS, args);
+
+        return output;
+    }
+
+    @Override
+    public Object visit(PropertyIsNil filter, Object extraData) {
+        ObjectNode output = asObjectNode(extraData);
+        output.put(OP, "isNull");
+
+        ArrayNode args = objectMapper.createArrayNode();
+        Expression expr = filter.getExpression();
+        expr.accept(new ExpressionToCQL2Json(objectMapper), args);
+        output.set(ARGS, args);
+
+        return output;
+    }
+
+    @Override
+    public Object visit(BBOX filter, Object extraData) {
+        throw unsupported("BBOX");
```

Comment:
Ouch, a BBOX filter is very common, it should be supported. Can be translated to an intersection filter if needed?

Reply:
Added as intersection

Reply:
Good. I don't see a test covering this code path, did I miss it?

Reply:
https://github.com/geotools/geotools/blob/c694004b414dc93c3a631de2e2f4e1a175ea461f/modules/unsupported/cql-json/src/test/java/org/geotools/filter/text/cqljson/FilterToJsonTest.java#L243